File: Ransom Note.cpp
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        map<char,int> m;
        for(int i=0;i<magazine.length();i++)
        {
            if(m.find(magazine[i])==m.end())
            {
                m[magazine[i]]=1;
            }
            else
            {
                m[magazine[i]]++;
            }
        }
        for(int i=0;i<ransomNote.length();i++)
        {
            if(m.find(ransomNote[i])==m.end())
            {
                return false;
            }
            else if(m[ransomNote[i]]==0)
            {
                return false;
            }
            else
            {
                m[ransomNote[i]]--;
            }
        }
        return true;
    }
};

File: Golang Solutions/Ransom Note.go
func canConstruct(ransomNote string, magazine string) bool {
    m:= make(map[byte]int)
    for i:=0;i<len(magazine);i++{
        
        _, ok := m[magazine[i]]
        if !ok {
            m[magazine[i]]=1
        }else{
            m[magazine[i]]++
        }
        
    }
    for i:=0;i<len(ransomNote);i++{
        val, ok := m[ransomNote[i]]
        if !ok {
            return false;
        }else if val==0{
            return false;   
        }else{
            m[ransomNote[i]]--;
        }
        
    }
    return true;
    
}